pkg/cluster/resources.go

```diff
@@ -142,3 +142,33 @@ func CreateNamespace(cli client.Client, namespace string, metaOptions ...MetaOpt
 
 	return foundNamespace, nil
 }
+
+// CreateClusterRole creates cluster role required by model registry component.
+func CreateClusterRole(cli client.Client, clusterRole string, rules []authv1.PolicyRule, metaOptions ...MetaOptions) (*authv1.ClusterRole, error) {
+	desiredClusterRole := &authv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: clusterRole,
+		},
+		Rules: rules,
+	}
+
+	if err := ApplyMetaOptions(desiredClusterRole, metaOptions...); err != nil {
+		return nil, err
+	}
+
+	foundClusterRole := &authv1.ClusterRole{}
+	err := cli.Get(context.TODO(), client.ObjectKey{Name: clusterRole}, foundClusterRole)
+	if err != nil {
+		if apierrs.IsNotFound(err) {
+			err = cli.Create(context.TODO(), desiredClusterRole)
+			if err != nil && !apierrs.IsAlreadyExists(err) {
+				return nil, err
+			}
+			desiredClusterRole.DeepCopyInto(foundClusterRole)
+		} else {
+			return nil, err
+		}
+	}
+
+	return foundClusterRole, nil
```

Comment:
A personal-taste suggestion. It might improve readability a bit as cyclomatic complexity is lower (lazy 🧠 can comprehend quicker)

```suggestion
	if getErr := cli.Get(context.TODO(), client.ObjectKey{Name: name}, foundClusterRole); client.IgnoreNotFound(getErr) != nil {
		return nil, getErr
	}

	if createErr := cli.Create(context.TODO(), desiredClusterRole); client.IgnoreAlreadyExists(createErr) != nil {
		return nil, createErr
	}

	desiredClusterRole.DeepCopyInto(foundClusterRole)
	return foundClusterRole, nil
```

> [!WARNING]
> Suggestion takes into account nit rename of `clusterRole` to `name`. So choose wisely ;)